code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -195,6 +195,7 @@
 			return TRUE // It's resolved. No afterattack needed. Stops you from emagging lockers when putting in an emag
 	else if(can_be_emaged && (istype(W, /obj/item/card/emag) || istype(W, /obj/item/melee/energy/blade) && !broken))
 		emag_act(user)
+		return TRUE
```

Comment:
did you mean to put this inside `emag_act()` ?

Reply:
Yup, will remove, thank you